CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/scatter3d/Scatter3dArranger.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2010-2020 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.arrangements.scatter3d;
+
+import au.gov.asd.tac.constellation.graph.Graph;
+import au.gov.asd.tac.constellation.graph.GraphElementType;
+import au.gov.asd.tac.constellation.graph.GraphWriteMethods;
+import au.gov.asd.tac.constellation.graph.schema.analytic.attribute.objects.RawData;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+import au.gov.asd.tac.constellation.plugins.arrangements.Arranger;
+import au.gov.asd.tac.constellation.plugins.arrangements.GraphUtilities;
+import au.gov.asd.tac.constellation.utilities.color.ConstellationColor;
+import java.time.ZonedDateTime;
+import java.util.BitSet;
+
+/**
+ * Arrange the graph in a scatter3d.
+ *
+ * @author CrucisGamma
+ */
+public class Scatter3dArranger implements Arranger {
+
+    private final Scatter3dChoiceParameters params;
+
+    /**
+     * A scatter3d arrangement with default parameters.
+     */
+    public Scatter3dArranger() {
+        this(Scatter3dChoiceParameters.getDefaultParameters());
+    }
+
+    /**
+     * Construct new ArrangeInScatter3d instance.
+     *
+     * @param params Parameters for the arrangement.
+     */
+    public Scatter3dArranger(final Scatter3dChoiceParameters params) {
+        this.params = params;
+    }
+
+    @Override
+    public void setMaintainMean(final boolean b) {
+    }
+
+    @Override
+    public void arrange(final GraphWriteMethods wg) throws InterruptedException {
+
+        // Get/set the x,y,z attributes.
+        if (wg.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.X.getName()) == Graph.NOT_FOUND) {
+            wg.addAttribute(GraphElementType.VERTEX, "float", "x", "x", null, null);
+        }
+        if (wg.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.Y.getName()) == Graph.NOT_FOUND) {
+            wg.addAttribute(GraphElementType.VERTEX, "float", "y", "y", null, null);
+        }
+        if (wg.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.Z.getName()) == Graph.NOT_FOUND) {
+            wg.addAttribute(GraphElementType.VERTEX, "float", "z", "z", null, null);
```

Comment:
change the 3 instances of to "float" to FloatAttributeDescription.getName()

Reply:
Copied from Grid.  Will change